crates/pera-core/src/dwallet_mpc/network_dkg.rs

```diff
@@ -36,67 +48,248 @@ pub struct DwalletMPCNetworkKeyVersions {
 /// Encapsulates all the fields in a single structure for atomic access.
 pub struct DwalletMPCNetworkKeyVersionsInner {
     /// The validators' decryption key shares.
-    pub validator_decryption_key_share: HashMap<DWalletMPCNetworkKey, Vec<Vec<u8>>>,
+    pub validator_decryption_key_share: HashMap<
+        DWalletMPCNetworkKeyScheme,
+        Vec<HashMap<PartyID, <AsyncProtocol as Protocol>::DecryptionKeyShare>>,
```

Comment:
When you don't take `AsyncProtocol` to be generic here, you lose the entire purpose of this structure; you make a structure that is supposed to support multiple types of keys, but you use the type of a single specific key for it. How can it work?

Reply:
Probably you should just save the `SecretKeyShareSizedNumber` here and create the decryption key share when you know the type for it